libs/sdk-common/src/input_parser.rs

```diff
@@ -222,6 +314,11 @@ async fn parse_core(input: &str) -> Result<InputType> {
         };
     }
 
+    #[cfg(feature = "liquid")]
```

Comment:
I don't think we should use the `extract_bolt12_offer` & `extract_lnurl` inside `parse_core`.
Currently this specific bip21 format (without the bitcoin) is only intended to be used as part of the bip 353 spec so I feel more comfortable of using this logic inside the `bip353_parse` function and return from there either the offer or the lnurl.

Also for the internal parsing (and extracting the either offer or lnurl) in `bip353_parse` we can use a logic similar to this one: https://github.com/FGRibreau/querystring/blob/master/src/lib.rs#L38

1. First extract the query part (after the ? )
2. Then use that logic to create a hash table of key/value
3. Look for the "lno" first and after that for the "lnurl" and return the value from  the bip353_parse function.

What do you think about this suggestion @lorenzoronzani ?